File: chap14/fig14-4.py
import pandas
from plotnine import ggplot, aes, theme_bw, theme, element_blank, geom_tile, scale_fill_gradient

N = 50
I = 120
d_ori = pandas.read_csv('input/data-matrix-decomp.csv')
d = d_ori.value_counts(sort=False).reset_index()
d.columns = ['PersonID', 'ItemID', 'Freq']
d = d[d.Freq >= 1].reset_index()

p = (ggplot(d, aes('ItemID', 'PersonID', fill='Freq'))
    + theme_bw(base_size=22)
    + theme(axis_text=element_blank(), axis_ticks=element_blank())
    + geom_tile()
    + scale_fill_gradient(breaks=[1,5,10], low='gray', high='black')
)
p.save(filename='output/fig14-4.py.png', dpi=300, width=8, height=6)

File: chap14/fig14-4.R
library(dplyr)
library(ggplot2)

N <- 50
I <- 120
d_ori <- read.csv(file='input/data-matrix-decomp.csv')
d <- data.frame(table(d_ori)) %>% filter(Freq >= 1)
d <- d %>% 
  mutate(PersonID = as.integer(PersonID),
         ItemID = as.integer(ItemID))

p <- ggplot(data=d, aes(x=ItemID, y=PersonID, fill=Freq)) +
  theme_bw(base_size=22) +
  theme(axis.text=element_blank(), axis.ticks=element_blank()) +
  geom_tile() +
  scale_fill_gradient(breaks=c(1,5,10), low='gray80', high='black')
ggsave(file='output/fig14-4.png', plot=p, dpi=300, w=8, h=6)
